internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -429,6 +429,29 @@ func (ctrl *ControlPlaneStaticPodController) manageAPIServer(ctx context.Context
 		"service-account-signing-key-file": argsbuilder.MergeDenied,
 		"tls-cert-file":                    argsbuilder.MergeDenied,
 		"tls-private-key-file":             argsbuilder.MergeDenied,
+		"authorization-config":             argsbuilder.MergeDenied,
+	}
+
+	k8sVersion := compatibility.VersionFromImageRef(cfg.Image)
+
+	if k8sVersion.KubeAPIServerSupportsAuthorizationConfigFile() {
+		if _, ok := cfg.ExtraArgs["authorization-mode"]; ok {
```

Comment:
I'm confused here a bit, if someone previously had `authorization-mode` set via extra args, I don't think we should stop supporting it, we should just keep it and (and same for `webhook-` args?)

Reply:
i guess we can check if both are set and during machine config validation, I was wondering what I forgot to check

Reply:
handled all cases and added tests